### ISSUE TITLE
Beets is haunted, reads info from comment lines?

### ISSUE THREAD

**Issue by RollingStar** [beetbox/beets#2565]
### Problem

A lot is going on here. May be many bugs combining? Some thoughts:

* Beets reading info to the right of a comment indicator `#` which is unintuitive
* Information in replace somehow ruining musicbrainz.host
* Maybe something to do with using git branch rather than latest stable (1.4.3)?
* Maybe something to do with latest commit?

Running this command in verbose (`-vv`) mode:

```sh
beet -vv
configuration error: musicbrainz.host not found
```

### Setup

* OS: win10 x64
* Python version: 3.6
* beets version: Latest git (https://github.com/beetbox/beets/commit/9840964f51fed3b3d0dcc1db9c76f90be4c05e73)
* Turning off plugins made problem go away (yes/no): no

My configuration (output of `beet config`) is:

Causes issue:

```yaml
library: e:\songs\beetslibrary.bib

replace:
    'qqqqqqq': 'eeeeeeeee'
    #”
```

Works fine:

```yaml
library: e:\songs\beetslibrary.bib

replace:
    'qqqqqqq': 'eeeeeeeee'
    #
```

**Comment by sampsyo** [comment]
Hmm; that's odd—this one I can't seem to reproduce.

```
$ cat config.yaml
library: e:\songs\beetslibrary.bib

replace:
    'qqqqqqq': 'eeeeeeeee'
    #”
$ beet -vv config
[...clip...]
Sending event: library_opened
verbose: 2
library: e:\songs\beetslibrary.bib

replace:
    qqqqqqq: eeeeeeeee

Sending event: cli_exit
```

Based on our experience last time, I also tried removing the newline from the end of the file—still no issue.

I'm kind of stumped. Maybe it has something to do with Windows?

**Comment by RollingStar** [comment]
Happens with or without the newline. Note the troublesome character is [U+201D](http://www.fileformat.info/info/unicode/char/201d/index.htm) and not a normal quote `"`. All these don't work.

```
library: e:\songs\beetslibrary.bib

replace:
    'q': '”'
```

```
library: e:\songs\beetslibrary.bib

paths:
    default: '”'
```

```
library: e:\songs\beetslibrary.bib

paths:
    default: ”
```

This works:

```
library: e:\songs\beetslibrary.bib

paths:
    default: a
```


So, maybe `”` is being treated as an escape character? If the text handler parses escape characters before comments `#` that would explain all the results thus far.

**Comment by RollingStar** [comment]
Okay. Applying my fix from #2456 fixes my issue with the quotes and DEL and everything else. However, could error handling be improved for other unforseeable errors?

* Beets reading info to the right of a comment indicator `#`. Is Beets reading line by line, or the entire file at once? Line by line would have a benefit of being more assured that beets is not reading any line after # appears.
* The "configuration error" that is thrown. What is happening between program defaults & user config that causes this error?

**Comment by sampsyo** [comment]
OK—that sounds like we're dealing with an encoding error.

I'm guessing that the YAML parser is somehow giving up entirely when it encounters a character it doesn't understand. I'm still at a loss to explain why that leads to the configuration error you're seeing—it seems to throw off the whole configuration system, rather than just exiting with a parse error. It's hard to narrow down what's wrong without a way to reproduce the problem.
